.cloudbuild/ci/build.yaml

```diff
@@ -18,17 +18,17 @@ steps:
       - "go run ./cmd/checkout -w=/workspace"
 
   - name: gcr.io/cloud-builders/docker
-    id: type-check-and-lint
+    id: prettier-check-and-lint
```

Comment:
Edit: nvm, I wrote this comment and then saw you addressed this in the next comment. ;f

---

With the new config instead of typechecking the code once we're going to check it twice, first when compiling the Web UI and then again when compiling Connect, right?

I don't think that's necessarily bad, one could even say it provides some additional granularity so that you can see your change broke the Web UI but not Connect.

---

tl;dr should this step run in parallel to other build steps now?

For me type checking was the most important part of `type-check-and-lint` – if the code doesn't type check then it's probably broken.

Then there's building which is another important step because if it doesn't build then it means that your PR broke something.

So one piece that we're left with is `prettier-check-and-lint` which now does just linting. IMHO most of the time that's a largely inconsequential check. I reeeally dislike when a minor problem with linting prevents the build from getting to the real deal which is type checking and actually attempting to build the code.

With that in mind, should we make it so that this step, renamed to lint, runs in parallel with `build-teleport` and `build-term`?

Reply:
Yeah, but the `ts-loader` is set only compile bundled files, so it shouldn't type check anything that doesn't end up in the main bundle. This means the web UI will only type check the web UI code, and Connect will only type check Connect code

Reply:
Also, I tried to change the `waitFor` from linting to checkout so they run in parallel, but it then builds the docker container three times - I'm not familiar with cloud build's config, so I didn't look into it much further and reverted the change

Reply:
I'll see if adding an empty intermediate step helps with that.

Reply:
Something like this should help with caching the docker image between steps:

https://github.com/gravitational/webapps/commit/c278ceca8b9f2c7ad1637b7433b204c3d435b2d9

https://console.cloud.google.com/cloud-build/builds;region=global/3911680d-bd1a-46c9-85d0-280faf69e9ef;step=2?project=ci-account